doc/release/upcoming_changes/15840.compatibility.rst

```diff
@@ -0,0 +1,7 @@
+Change output of ``__round__`` to be consistent with Python
+-----------------------------------------------------------
+
+Output of the ``__round__`` dunder method and consequently the Python
+built-in ``round`` has been changed to be a Python ``int`` to be consistent
+with calling it on Python ``float`` objects. Previously, it would return
+a scalar of the ``dtype`` that was passed in.
```

Comment:
Could try and rephrase it to include the `default` (integral) rounding has been changed... But, it is probably fine and I am not quite sure how to phrase it better right now. The "default" word is the only idea I have right now.